component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java

```diff
@@ -1153,6 +1195,60 @@ public Response postActivityOnSpace(@Context UriInfo uriInfo,
 
     return EntityBuilder.getResponse(EntityBuilder.buildEntityFromActivity(activity, uriInfo.getPath(), expand), uriInfo, RestUtils.getJsonMediaType(), Response.Status.OK);
   }
+  
+  /**
+   * {@inheritDoc}
+   */
+  @POST
+  @Path("activities/{activityId}/share")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Shares the given activity to specific spaces",
+                httpMethod = "POST",
+                response = Response.class,
+                notes = "This shares the given activity to the given spaces if the authenticated user is a member of the target not redactional space or a super admin, a spaces super manager, a redactor or a manager of the target redactional space.")
+  @ApiResponses(value = {
+    @ApiResponse (code = 200, message = "Request fulfilled"),
+    @ApiResponse (code = 500, message = "Internal server error"),
+    @ApiResponse (code = 400, message = "Invalid query input") })
+  public Response shareActivityOnSpaces(@Context UriInfo uriInfo,
+                                      @ApiParam(value = "Activity id to be shared", required = true) @PathParam("activityId") String activityId,
+                                      @ApiParam(value = "Asking for a full representation of a specific subresource, ex: comments or likes", required = false) @QueryParam("expand") String expand,
+                                      @ApiParam(value = "Share target spaces", required = true) SharedActivityRestIn sharedActivityRestIn) throws Exception {
+    String authenticatedUser = ConversationState.getCurrent().getIdentity().getUserId();
+    ActivityManager activityManager = CommonsUtils.getService(ActivityManager.class);
+    if (activityManager.getActivity(activityId) == null) {
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    if (sharedActivityRestIn == null || sharedActivityRestIn.getTargetSpaces() == null || sharedActivityRestIn.getTargetSpaces().isEmpty() || sharedActivityRestIn.getType() == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    
+    Identity authenticatedUserIdentity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, authenticatedUser);
+    List<ActivityEntity> sharedActivitiesEntities = new ArrayList<ActivityEntity>();
+    for (String targetSpaceName : sharedActivityRestIn.getTargetSpaces()) {
+      Space targetSpace = spaceService.getSpaceByPrettyName(targetSpaceName);
+      if (SpaceUtils.isSpaceManagerOrSuperManager(authenticatedUser, targetSpace.getGroupId()) || (spaceService.isMember(targetSpace, authenticatedUser) && SpaceUtils.isRedactor(authenticatedUser, targetSpace.getGroupId()))) {
```

Comment:
this is silently ignoring attempts to share in spaces where user is not supposed to.
first it's questionable wether HTTP 200 should be used in such situation since, it was not fully completed (you have no way to tell the user that the share didn't work on all spaces). But more importantly  IMO, we should at least log it server side as it may be some kind of attack or denoting an inconsistency bug